### DISCUSSION TITLE
Lenses does not respect `hideFromIndex` when field is in Stack

### DISCUSSION THREAD

**Discussion by webard** [laravel/nova-issues#6444] [Q&A] [answered]
- Laravel Version: 11.1.11
- Nova Version: 4.34.3
- PHP Version: 8.3.8
- Database Driver & Version: MySQL 8.3.0
- Operating System and Version: MacOS 14.5 (23F79)
- Browser type and version: Chrome 126.0.6478.57 

### Description:

```php
Stack::make('Users', [
    BelongsTo::make('Patron', 'patron', User::class)
           ->hideFromIndex(function ($request, $resource) {
                 return  $resource->type === 'incoming';
        })
   });
```

When any field is in `Stack` field in Lenses, hideFromIndex is never called. hideFromIndex works great when there is no `Stack`, and `Stack` fields with `hideFromIndex` works correctly on resource page. Only Lens page has problem.

What I want to do is hide field on index depending on other field value.

### Detailed steps to reproduce the issue on a fresh Nova installation:
1. Create Stack in Lenses
2. Create Field in Stack
3. Make `hideFromIndex` with checking something, or just return true

**Comment by crynobone** [comment]
Shouldn't you already be able to use `canSee()` here?

**Reply by webard** [reply]
@crynobone canSee does not work on tables, there is no access to data, it is known problem.

**Reply by crynobone** [reply, marked as answer]
Lenses can only display index so having an option to hide it isn't ideal.
